vernac/comArguments.ml

```diff
@@ -183,6 +183,14 @@ let vernac_arguments ~section_local reference args more_implicits flags =
 
   let names = List.fold_left names_union [] names in
 
+  let duplicate_names =
+    List.duplicates Name.equal (List.filter ((!=) Anonymous) names)
+  in
+  if not (List.is_empty duplicate_names) then begin
+    CErrors.user_err Pp.(strbrk "Some argument names are duplicated: " ++
+                         prlist_with_sep pr_comma Name.print duplicate_names)
+  end;
+
```

Comment:
Not a big deal, but why the move?

Reply:
After the move, `Arguments` only checks for duplicates in the explicitly user-given names, not in the final list of names.
For example, without this change:
```coq
Definition foo (a a : nat) := a.
Arguments foo _ _ : assert.
```
fails with `Some argument names are dupliated: a`, and this happened several places in the CI.

I'm not entirely happy with this change, as it probably lets through some unexpected corner cases, but it is a minimal change that reduces incompatibility with existing developments, and doesn't completely get rid of the (useful) error.
The `Arguments` command is probably due to be looked at closely sometime soon, but maybe after more of your changes to implicits go in.

Reply:
Since it is now allowed internally, why not let the user repeat names in `Arguments`?

Reply:
My reasoning was that since there is still no way to *use* repeated names in `(name := arg)` applications, users almost never want duplicated names and so the error is useful. However, taking another look, that seems overly paternalistic; names can also simply be documentation, and we allow naming non-implicit arguments. Since the check for duplicates is already incomplete, it is probably better to be more consistent and remove this check.